core/desugarDatatypes.ml

```diff
@@ -209,6 +213,72 @@ module Desugar = struct
     let seed =
       let open Datatype in
       match rv with
+        | EffectApplication (name, ts) ->
+            let match_quantifiers : type a. (a -> Kind.t) -> a list -> Types.type_arg list = fun proj qs ->
+              let match_kinds i (q, t) =
+                let primary_kind_of_type_arg : Datatype.type_arg -> PrimaryKind.t = function
+                  | Type _ -> PrimaryKind.Type
+                  | Row _ -> PrimaryKind.Row
+                  | Presence _ -> PrimaryKind.Presence
+                in
+                let q_kind, _ = proj q in
+                let t_kind = primary_kind_of_type_arg t in
+                if q_kind <> t_kind then
+                  raise
+                    (TypeApplicationKindMismatch
```

Comment:
`TypeApplicationKindMismatch` should probably be defined in `errors.ml`.

Reply:
yes it is, what should I do there ?